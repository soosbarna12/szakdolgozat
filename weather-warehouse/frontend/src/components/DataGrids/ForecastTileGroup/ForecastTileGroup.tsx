import Grid from "@mui/material/Grid2";
import React from "react";
import { DAYS } from "../../../consts/days";
import ForecastTile from "./ForecastTile/ForecastTile";

export default function ForecastTileGroup() {

  function renderForecastTiles() {
    return DAYS.map((day) => (
      <ForecastTile key={day.id} dayName={day.text} data-testid="forecastTile" />
    ))
  };


  return (
    <Grid container columns={7} spacing={2} justifyContent="center" alignItems="center" data-testid="forecastTileGroup" >
      {renderForecastTiles()}
    </Grid >
  );
}