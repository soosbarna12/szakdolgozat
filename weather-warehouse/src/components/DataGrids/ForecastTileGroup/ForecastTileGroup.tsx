import Grid from "@mui/material/Grid2";
import React from "react";
import { DAYS } from "../../../consts/days";
import ForecastTile from "./ForecastTile/ForecastTile";

export default function ForecastTileGroup() {

  function renderForecastTiles() {
    return DAYS.map((day) => (
      <ForecastTile key={day.id} dayName={day.text} />
    ))
  };


  return (
    <Grid container columns={7} spacing={2} justifyContent="center" alignItems="center">
      {renderForecastTiles()}
    </Grid>
  );
}