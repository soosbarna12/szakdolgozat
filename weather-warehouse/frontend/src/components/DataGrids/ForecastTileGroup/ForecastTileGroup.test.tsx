import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import React from 'react';
import { renderWithQueryClient } from '../../../utils/test/renderWithQueryClient';
import ForecastTileGroup from './ForecastTileGroup';

jest.mock('react-leaflet', () => ({
    MapContainer: ({ children }: { children: React.ReactNode }) => (
        <div data-testid="dataMapContainer">{children}</div>
    ),
}));

describe('DataGrids/ForecastTileGroup', () => {
    it('matches ForecastTileGroup component snapshot', () => {
        const component = renderWithQueryClient(<ForecastTileGroup />);
        expect(component).toMatchSnapshot();
    });

    it('renders all forecast tiles with correct day names', () => {
        renderWithQueryClient(<ForecastTileGroup />);

        const forecastTiles = screen.getAllByTestId('forecastTile');
        expect(forecastTiles).toHaveLength(7); // Assuming there are 7 days in DAYS
    });
});