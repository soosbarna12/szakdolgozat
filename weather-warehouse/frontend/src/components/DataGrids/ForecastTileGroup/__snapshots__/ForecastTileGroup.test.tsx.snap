// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrids/ForecastTileGroup matches ForecastTileGroup component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-128divp-MuiGrid2-root"
        data-testid="forecastTileGroup"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
          data-testid="forecastTile"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          Monday
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
          data-testid="forecastTile"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          Tuesday
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
          data-testid="forecastTile"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          Wednesday
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
          data-testid="forecastTile"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          Thursday
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
          data-testid="forecastTile"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          Friday
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
          data-testid="forecastTile"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          Saturday
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
          data-testid="forecastTile"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          Sunday
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-128divp-MuiGrid2-root"
      data-testid="forecastTileGroup"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
        data-testid="forecastTile"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        Monday
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
        data-testid="forecastTile"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        Tuesday
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
        data-testid="forecastTile"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        Wednesday
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
        data-testid="forecastTile"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        Thursday
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
        data-testid="forecastTile"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        Friday
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
        data-testid="forecastTile"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        Saturday
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3uc5wr-MuiPaper-root"
        data-testid="forecastTile"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        Sunday
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
